File: 2nd year/Laboratories/MAP/7/src/main/java/ui/tui/TaskTUI.java
package ui.tui;

import service.TaskService;

import java.util.HashMap;
import java.util.Map;
import java.util.stream.Stream;

class TaskTUI extends BaseTUI {
    private TaskService taskService;

    TaskTUI(TaskService taskService) {
        this.taskService = taskService;
        generateTUI(
                "Task TUI",
                new HashMap<String, Runnable>() {{
                    put("Create task", TaskTUI.this::createTask);
                    put("Read one task", TaskTUI.this::readOneTask);
                    put("Read all tasks", TaskTUI.this::readAllTasks);
                    put("Update task", TaskTUI.this::updateTask);
                    put("Delete task", TaskTUI.this::deleteTask);
                    put("Extend task deadline", TaskTUI.this::extendTaskDeadline);
                }});
    }

    private void createTask() {
        Map<String, String> taskMap = readMap("id", "assignmentWeek", "deadlineWeek", "description");
        System.console().printf("Operation %s\n", taskService.createTask(taskMap) ? "successful" : "failed");
    }

    private void readOneTask() {
        String taskId = readOne("id");
        Stream.of(taskService.readOneTask(taskId))
                .filter(s -> !s.isEmpty())
                .forEach(s -> System.console().printf("%s\n", s));
    }

    private void readAllTasks() {
        taskService.readAllTasks()
                .forEach(s -> System.console().printf("%s\n", s));
    }

    private void updateTask() {
        Map<String, String> taskMap = readMap("id", "assignmentWeek", "deadlineWeek", "description");
        System.console().printf("Operation %s\n", taskService.updateTask(taskMap) ? "successful" : "failed");
    }

    private void deleteTask() {
        String taskId = readOne("id");
        System.console().printf("Operation %s\n", taskService.deleteTask(taskId) ? "successful" : "failed");
    }

    private void extendTaskDeadline() {
        String taskId = readOne("taskId");
        String newDeadlineWeek = readOne("newDeadlineWeek");
        System.console().printf("Operation %s\n", taskService.extendTaskDeadline(taskId, newDeadlineWeek) ? "successful" : "failed");
    }
}

File: 2nd year/Laboratories/MAP/L9/CRUD/UI/Tui/TaskTui.cs
using System;
using System.Collections.Generic;
using CRUD.Domain;
using CRUD.Service;
using CRUD.Utils;

namespace CRUD.UI.Tui
{
    public class TaskTui : BaseTui
    {
        private readonly TaskService _service;

        public TaskTui(TaskService service)
        {
            _service = service;
            Initialize("Task TUI", new Dictionary<string, Action>
            {
                {"Create task", CreateTask},
                {"Update task", UpdateTask},
                {"Delete task", DeleteTask},
                {"Read one task", ReadOneTask},
                {"Read all tasks", ReadAllTasks},
                {"Extend task deadline", ExtendTaskDeadline}
            });
        }

        private void CreateTask()
        {
            var task = Read("Id", "AssignmentWeek", "DeadlineWeek", "Description");
            Write(GenerateMessage(_service.CreateTask(task)));
        }

        private void UpdateTask()
        {
            var task = Read("Id", "AssignmentWeek", "DeadlineWeek", "Description");
            Write(GenerateMessage(_service.UpdateTask(task)));
        }

        private void DeleteTask()
        {
            var id = Read("Id");
            Write(GenerateMessage(_service.DeleteTask(id)));
        }

        private void ReadOneTask()
        {
            var id = Read("Id");
            var task = _service.ReadOneTask(id);
            Write(task == null
                ? "Task not found..."
                : EntityUtils.DumpString<Task, int>(task));
        }

        private void ReadAllTasks()
        {
            foreach (var task in _service.ReadAllTasks())
            {
                Write(EntityUtils.DumpString<Task, int>(task));
            }
        }

        private void ExtendTaskDeadline()
        {
            var id = Read("Id");
            var newDeadlineWeek = Read("NewDeadlineWeek");
            Write(GenerateMessage(_service.ExtendTaskDeadline(id, newDeadlineWeek)));
        }
    }
}